block-producer/src/config.rs

```diff
@@ -1,58 +1,94 @@
-use std::fmt::Display;
-
+use miden_node_store::config::StoreConfig;
+use miden_node_utils::config::{Config, HostPort};
 use serde::{Deserialize, Serialize};
 
 pub const HOST: &str = "localhost";
 // defined as: sum(ord(c)**p for (p, c) in enumerate('miden-block-producer', 1)) % 2**16
 pub const PORT: u16 = 48046;
-pub const ENV_PREFIX: &str = "MIDEN_BLOCK_PRODUCER";
 pub const CONFIG_FILENAME: &str = "miden-block-producer.toml";
 
-#[derive(Clone, Eq, PartialEq, Ord, PartialOrd, Hash, Debug, Serialize, Deserialize)]
-pub struct Endpoint {
-    /// The host device the server will bind to.
-    pub host: String,
-    /// The port number to bind the server to.
-    pub port: u16,
-}
+// Main config
+// ================================================================================================
 
+/// Block producer specific configuration
 #[derive(Clone, Eq, PartialEq, Ord, PartialOrd, Hash, Debug, Serialize, Deserialize)]
 pub struct BlockProducerConfig {
-    /// Main endpoint of the server.
-    pub endpoint: Endpoint,
+    pub host_port: HostPort,
 
-    /// Endpoint of the store gRPC server
+    /// Store gRPC endpoint in the format `http://<host>[:<port>]`.
     pub store_endpoint: String,
 }
 
-impl Default for Endpoint {
+impl Default for BlockProducerConfig {
     fn default() -> Self {
         Self {
-            host: HOST.to_string(),
-            port: PORT,
+            host_port: HostPort {
+                host: HOST.to_string(),
+                port: PORT,
+            },
+            store_endpoint: StoreConfig::default().as_endpoint(),
         }
     }
 }
 
-impl Display for Endpoint {
-    fn fmt(
-        &self,
-        f: &mut std::fmt::Formatter<'_>,
-    ) -> std::fmt::Result {
-        write!(f, "{}:{}", self.host, self.port)
+impl BlockProducerConfig {
+    pub fn as_endpoint(&self) -> String {
+        format!("http://{}:{}", self.host_port.host, self.host_port.port)
     }
```

Comment:
naming nit: I'd probably name this `get_endpoint()` or maybe even use "url" instead of "endpoint" - e.g., ` get_url()`.